scapy/supersocket.py

```diff
@@ -189,6 +189,8 @@ class L2ListenTcpdump(SuperSocket):
 
     def __init__(self, iface=None, promisc=None, filter=None, nofilter=False,
                  prog=None, *arg, **karg):
+        if DARWIN:
+            raise OSError("L2ListenTcpdump is not available on OSX because tcpdump does not support piped input.")
```

Comment:
For more info, see
https://github.com/secdev/scapy/blob/master/scapy/utils.py#L1276-L1292

Reply:
I don't understand. This should be handled under Darwin by the code you are linking, shouldn't it?

Reply:
Well you're right :o just don't understand why this very test fails on OSX only :/ (the rest is working, i've rebased)